apps/src/netsim/netsimUtils.js

```diff
@@ -207,6 +207,28 @@ exports.deserializeNumber = function (storedNum) {
 };
 
 /**
+ * Helper for converting from an older header-spec format to a new, simpler one.
+ * Old format: {key:{string}, bits:{number}}[]
+ * New format: string[]
+ * If we detect the old format, we return a spec in the new format.
+ * @param {Array} spec
+ * @returns {Array}
+ */
+exports.scrubHeaderSpecForBackwardsCompatibility = function (spec) {
+  var scrubbedSpec = [];
+  spec.forEach(function (specEntry) {
+    if (typeof specEntry === 'string') {
+      // This is new new format, we can just copy it over.
+      scrubbedSpec.push(specEntry);
+    } else if (specEntry !== null && typeof specEntry === 'object') {
+      // This is the old {key:'', bits:0} format.  We just want the key.
+      scrubbedSpec.push(specEntry.key);
```

Comment:
An approach Brian took in a similar situation (where we were converting format of level data) was to either console.log or .warn when we saw old data. This meant we would see when we had old data (and have motivation to convert the level so that we wouldn't see said message), and also have a sense of how much old data was out there. Eventually, when we were confident we'd converted everything, we removed the logging and special handling. Not sure if it could make sense to do something similar here